com.unity.multiplayer.mlapi/Runtime/Core/NetworkManager.cs

```diff
@@ -738,6 +730,13 @@ private void OnEnable()
             {
                 Application.runInBackground = true;
             }
+
+            if (Singleton != null && Singleton != this)
+            {
+                return;
+            }
+
+            SetSingleton();
         }
 
         private void OnDestroy()
```

Comment:
I can't comment on the code below but we should also do this below:

```cs
Shutdown();

if (Singleton == this)
{
    Singleton = null;
}
```

otherwise, there'll be a bug — second `NetworkManager` instance won't call `Shutdown()` on itself on `OnDestroy()`, because it won't be the singleton instance.